lib/utils/repoGitDiff.js

```diff
@@ -1,24 +1,34 @@
 'use strict'
-const childProcess = require('child_process')
 const cpUtils = require('./childProcessUtils')
+const childProcess = require('child_process')
 const gc = require('./gitConstants')
+const fs = require('fs')
+const ignore = require('ignore')
 const os = require('os')
 const path = require('path')
 
+
 const fullDiffParams = ['diff', '--name-status', '--no-renames']
+const ig = ignore()
 
 module.exports = (config, metadata) => {
   const { stdout: diff } = childProcess.spawnSync(
     'git',
     [...fullDiffParams, config.from, config.to],
     { cwd: config.repo, encoding: gc.UTF8_ENCODING }
   )
+
+  if (config.ignore && fs.existsSync(config.ignore)) {
+    ig.add(fs.readFileSync(config.ignore).toString())
+  }
+
   return treatResult(cpUtils.treatDataFromSpawn(diff), metadata)
 }
 
 const treatResult = (repoDiffResult, metadata) =>
   repoDiffResult
     .split(os.EOL)
+    .filter(line => line && !ig.ignores(line))
```

Comment:
in fact we should use [double negation](https://riptutorial.com/javascript/example/3047/double-negation----x-) check here for the code to be more robust.

Could you change it to 
`
.filter(line => !!line && !ig.ignores(line))
`